src/utils/telemetry.ts

```diff
@@ -207,3 +225,43 @@ export const consoleTelemetryExporter: TelemetryExporter = {
     console.debug('Telemetry Count', count);
   },
 };
+
+/**
+ * Telemetry entry point to initialize the singleton {@link TelemetryProvider} or
+ * get the current {@link TelemetryProvider}
+ */
+export class Telemetry {
+  private static instance: TelemetryProvider | undefined;
+
+  /**
+   * Gets the singleton {@link TelemetryProvider} configured for the Data Layer Observer instance.
+   * Initializes the singleton {@link TelemetryProvider} on first call
+   *
+   * @param provider The {@link TelemetryProvider} to initialize; only effective on first call
+   * @param exporter The {@link TelemetryExporter} to use with the default Telemetry Provider; only
+   * used if no {@link TelemetryProvider} is given and only effective on first call
+   */
+  static getInstance(provider?: TelemetryProvider, exporter?: TelemetryExporter): TelemetryProvider {
```

Comment:
Just a thought: rather than introduce side effects with these parameters, I wonder if it would be better to introfcut two new static methods: `withExporter` and `withProvider`?

then `getInstance` will take no parameters and return the default provider + exporter if neither of the `with` methods are called.

WDYT?

Reply:
Overall I like this idea. How would you propose setting the default instance initially? I agree it's a little odd that `getInstance` is responsible for returning an instance and initially setting the instance -- but only if it's the first call. On subsequent calls, provider and exporter args are ignored.

Maybe we get away from the idea of silently setting the default instance and make it explicit. And, you can change the default instance if you wish (would be useful in tests). Example:

```
Telemetry.setInstance(Telemetry.withProvider(window._dlo_telemetryProvider));
```
or...
```
Telemetry.setInstance(Telemetry.withExporter(window._dlo_telemetryExporter));
```

We don't allow setting both a provider and an exporter, so it's up to the caller (`init` in our case) to determine which call to make. While the caller burden isn't ideal, it does make the inability to pass both a provider and an exporter more explicit.

If we lay out a new pattern here, it might be valuable to consider extending the new pattern to logging in the future. Example:

``
Logger.setInstance(Logger.withAppender('console'));
``

Reply:
I like that! 

`Telemetry.setInstance(Telemetry.with_X_())` reads nicely. Love the idea of extending the patter to other static accessors too.